core/src/main/java/org/infinispan/interceptors/distribution/ScatteredDistributionInterceptor.java

```diff
@@ -388,29 +394,52 @@ protected void scheduleKeyInvalidation(Object key, EntryVersion nextVersion, boo
       svm.scheduleKeyInvalidation(key, nextVersion, removed);
    }
 
-   private CompletionStage<Void> commitSingleEntryIfNewer(RepeatableReadEntry entry, InvocationContext ctx, VisitableCommand command) {
+   private CompletionStage<Void> commitOperation(RepeatableReadEntry entry, Predicate<RepeatableReadEntry> wasCommitted,
```

Comment:
Nothing defines `Predicate` as being side effect free. I think because streams has assumed that for its `Predicate`s that we just kinda assume that all over. I can try to think of a better name, maybe.